eo-maven-plugin/src/main/java/org/eolang/maven/hash/ChResolve.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.hash;
+
+import java.nio.file.Path;
+
+/**
+ * Resolve {@link CommitHash} depending on the
+ * parameters.
+ *
+ * @since 0.28.14
+ */
+public final class ChResolve implements CommitHash {
```

Comment:
@MikhailLipanin I'm not sure about it, but probably that class is configurable, which is not a good think to do. Please, read that article https://www.yegor256.com/2016/04/19/object-must-not-be-configurable.html

Reply:
@volodya-lombrozo, Idk, but look:
![image](https://user-images.githubusercontent.com/95661085/209630068-d8525594-ca3a-4b95-8797-66a11859c7e1.png)

Here is branching (behavior of `html` method depends on its `Encapsulated property`)

Reply:
@volodya-lombrozo Should I make a `@todo` ?

Reply:
@MikhailLipanin It's up to you to decide: whether you solve it in that PR or later.

Reply:
added todo